codecs/text/src/main/java/com/datastax/oss/dsbulk/codecs/text/string/StringConvertingCodec.java

```diff
@@ -66,9 +66,9 @@ protected boolean isNullOrEmpty(String s) {
    * The string to use when formatting internal inputs.
    *
    * <p>According to the documentation of <code>codec.nullStrings</code>, we must use the first
-   * string specified there, or the empty string, if that setting is empty.
+   * string specified there, or simply return null, if that setting is empty.
    */
   protected String nullString() {
-    return nullStrings.isEmpty() ? "" : nullStrings.get(0);
+    return nullStrings.isEmpty() ? null : nullStrings.get(0);
```

Comment:
IOW we are deferring the handling of nulls from the codec layer to the layer below, that is, the connector. I think it makes a lot fo sense to let the connector decide how to handle nulls, and also empty strings.